test/e2e/gameserver_test.go

```diff
@@ -260,12 +273,12 @@ func TestGameServerRestartBeforeReadyCrash(t *testing.T) {
 	messageAndWait := func(gs *agonesv1.GameServer, msg string, check func(gs *agonesv1.GameServer, pod *corev1.Pod) bool) error {
 		return wait.PollImmediate(3*time.Second, 3*time.Minute, func() (bool, error) {
 			gs, err := gsClient.Get(gs.ObjectMeta.Name, metav1.GetOptions{})
-			if err != nil {
+			if !assert.NoError(t, err) {
```

Comment:
Not sure this adds any value? (same with one below)

We're already capturing this below.